### ISSUE TITLE
 Failed global initialization: FileNotOpen: Failed to open "/opt/bitnami/mongodb/logs/mongodb.log"

### ISSUE THREAD

**Issue by adelowo** [bitnami/bitnami-docker-mongodb#231]
**Description**

MongoDB fails to start when running with docker on macOS

```
mongodb-primary_1    | 2020-09-02T10:44:49.447+0000 F CONTROL  [main] Failed global initialization: FileNotOpen: Failed to open "/opt/bitnami/mongodb/logs/mongodb.log"
```

**Steps to reproduce the issue:**

I am making use of this `docker-compose.yml` file

```
version: "3.7"

services:
  mongodb-primary:
    image: "bitnami/mongodb:4.0"
    user: root
    environment:
      - MONGODB_ADVERTISED_HOSTNAME=mongodb-primary.app-dev
      - MONGODB_PORT_NUMBER=37017
      - MONGODB_REPLICA_SET_MODE=primary
      - MONGODB_ROOT_PASSWORD=abc123
      - MONGODB_REPLICA_SET_KEY=replicasetkey123
    ports:
      - "37017:37017"
    volumes:
      - "mongodb_master_data:/bitnami"
    networks:
      localmongo:
        aliases:
          - "mongodb-primary.app-dev"

  mongodb-secondary:
    image: "bitnami/mongodb:4.0"
    user: root
    depends_on:
      - mongodb-primary
    environment:
      - MONGODB_ADVERTISED_HOSTNAME=mongodb-secondary.app-dev
      - MONGODB_PORT_NUMBER=37018
      - MONGODB_REPLICA_SET_MODE=secondary
      - MONGODB_PRIMARY_HOST=mongodb-primary.app-dev
      - MONGODB_PRIMARY_PORT_NUMBER=37017
      - MONGODB_PRIMARY_ROOT_PASSWORD=abc123
      - MONGODB_REPLICA_SET_KEY=replicasetkey123
    ports:
      - "37018:37018"
    networks:
      localmongo:
        aliases:
          - "mongodb-secondary.app-dev"

  mongodb-arbiter:
    image: "bitnami/mongodb:4.0"
    user: root
    depends_on:
      - mongodb-primary
    environment:
      - MONGODB_ADVERTISED_HOSTNAME=mongodb-arbiter.app-dev
      - MONGODB_PORT_NUMBER=37019
      - MONGODB_REPLICA_SET_MODE=arbiter
      - MONGODB_PRIMARY_HOST=mongodb-primary.app-dev
      - MONGODB_PRIMARY_PORT_NUMBER=37017
      - MONGODB_PRIMARY_ROOT_PASSWORD=abc123
      - MONGODB_REPLICA_SET_KEY=replicasetkey123
    ports:
      - "37019:37019"
    networks:
      localmongo:
        aliases:
          - "mongodb-arbiter.app-dev"

volumes:
  mongodb_master_data:
    driver: local

networks:
  localmongo:
    driver: bridge

```

**Describe the results you received:**

```
mongodb-secondary_1  |
mongodb-secondary_1  | mongodb 10:51:04.67 INFO  ==> ** MongoDB setup finished! **
mongodb-secondary_1  | mongodb 10:51:04.69 INFO  ==> ** Starting MongoDB **
mongodb-secondary_1  | 2020-09-02T10:51:04.723+0000 F CONTROL  [main] Failed global initialization: FileNotOpen: Failed to open "/opt/bitnami/mongodb/logs/mongodb.log"
localmongo_mongodb-arbiter_1 exited with code 1
```

**Describe the results you expected:**
Running container image

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version**

- Output of `docker version`:

```

Client: Docker Engine - Community
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:41:33 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a66213fe
  Built:            Mon Jun 22 15:49:27 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 7
  Running: 2
  Paused: 0
  Stopped: 5
 Images: 5
 Server Version: 19.03.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 1.944GiB
 Name: docker-desktop
 ID: CABZ:IGG6:522V:QNTQ:CWKV:A3U2:P7SA:JAQV:MODL:UWE6:JPM2:O7OM
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 58
  Goroutines: 67
  System Time: 2020-09-02T12:30:29.779401531Z
  EventsListeners: 4
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.26.2, build eefe0d31
docker-py version: 4.2.2
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by joancafom** [comment]
Hi @adelowo 

Thanks for your issue! I was successfully able to reproduce your problem. I will try to fix this and keep you posted 😄  In the meantime, you could try removing the lines `user: root` in your `docker-compose.yml` file (if that is not a problem for your needs) and that should solve the problem by now. Could you try that?

Thanks!

**Comment by adelowo** [comment]
Thanks @joancafom . It worked!!!.

**Comment by joancafom** [comment]
Hi @adelowo 

Just to let you know that this has been solved in our latest release `4.2.9-debian-10-r20`. You should be able to use `user: root` again. See https://github.com/bitnami/bitnami-docker-mongodb/commit/385762c16f1e531a478509906eb32130196c3226 .

Thanks for your issue!
